### PR TITLE
Improve CategorySearch in web

### LINKED ISSUE #814: More than 2 suggested categories in categorySearch?
**Issue Type:**

I think this falls into the category of enhancement 

**Platform:**

Web

**Description:**

Is there currently a way, or is it possible for you to add more than just two suggested categories in the categorySearch?

screen shot 2019-02-01 at 4 53 55 pm

### PR DESCRIPTION
**Pull request by kvin97**
fix #814

### CODE REVIEW

File: packages/web/src/components/search/CategorySearch.js [hunk 1]
```diff
@@ -858,32 +859,25 @@ class CategorySearch extends Component {
...
-				{
-					label: `${this.state.currentValue} in all categories`,
-					value: this.state.currentValue,
-					category: '*',
-					// no source object exists for category based suggestions
-					source: null,
-				},
-				{
-					label: `${this.state.currentValue} in ${filteredCategories[0].key}`,
+			const categorySuggestions = [];
```
Comment (jyash97) [inline packages/web/src/components/search/CategorySearch.js:862]: How about slicing the array of `filteredCategories` and then mapping over it. Something like this: 

```
const size = filteredCategories.length < categorySuggestionsListSize ?  filteredCategories.length  : categorySuggestionsListSize;
const categoriesToDisplay = filteredCatogories.slice(0, size).map((item) => {
   // push to array ( try using spread operator if possible )
});
```

Comment (jyash97) [reply packages/web/src/components/search/CategorySearch.js:862]: Also please try to test it locally rigorously.

**Comment by jyash97** [comment]
Can you also follow the PR template? They are present for a reason so that we don't miss documentation or other things related to PR.

**Comment by bietkul** [comment]
@jyash97 Can you review it?

File: packages/web/src/components/search/CategorySearch.js [hunk 2]
```diff
@@ -993,8 +988,9 @@ class CategorySearch extends Component {
 										...rest,
 									})}
 								{!this.hasCustomRenderer
-									&& isOpen
-									&& finalSuggestionsList.length ? (
+								&& isOpen
+								&& finalSuggestionsList.length ? (
+										// eslint-disable-next-line react/jsx-indent
```
Comment (jyash97) [inline packages/web/src/components/search/CategorySearch.js]: I think we wont need to disable eslint. Any reason for this ?

**Review by jyash97 — CHANGES_REQUESTED** [review]
LGTM 💯 

The only change we need is to revert eslint changes and the prop value will be a maximum of 10. We cant have more than that. You will need to change Prop type

**Comment by kvin97** [comment]
I found the following error with the removal of disable eslint.
fix1
Once I fixed it, following lint errors popped up.
fix2
Both errors comes as `eslint(react/jsx-indent)`.
I found similar kind of issue here.
https://github.com/yannickcr/eslint-plugin-react/issues/1679

**Comment by kvin97** [comment]
> LGTM 
> 
> The only change we need is to revert eslint changes and the prop value will be a maximum of 10. We cant have more than that. You will need to change Prop type

**Comment by kvin97** [comment]
shall I use the following propType to specify the range, by importing PropTypes as `prop-types`.
`categorySuggestionsListSize: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),`

**Comment by jyash97** [comment]
I think there is some eslint config issue on your side as I tried changing things on the same file and could not replicate. 

Also for `propTypes` you may need to define a custom prop type.

**Comment by kvin97** [comment]
To remove the eslint issue I revert back to a previous commit and move the code changes again. 
For `propTypes` custom prop type was defined. 
And I tested it locally rigorously.

**Pull request closed without merging**